prqlc/prqlc-parser/src/lexer/chumsky_0_10.rs

```diff
@@ -0,0 +1,793 @@
+}
+
+fn quoted_triple_string<'a>(
+    _escaped: bool, // Not used in this implementation
+) -> impl Parser<'a, ParserInput<'a>, Vec<char>, ParserError<'a>> {
+    // Helper function to create triple quoted string parsers
+    fn triple_quoted_parser<'p>(
+        quote: char,
+    ) -> impl Parser<'p, ParserInput<'p>, Vec<char>, ParserError<'p>> {
+        let triple_quote_open = just(quote).then(just(quote)).then(just(quote));
+        let triple_quote_close = just(quote).then(just(quote)).then(just(quote));
+
+        triple_quote_open
+            .ignore_then(
+                // Keep consuming characters until we hit three quotes in a row
+                any()
+                    .filter(move |&c| c != quote)
+                    .repeated()
+                    .collect::<Vec<char>>(),
+            )
+            .then_ignore(triple_quote_close)
+    }
+
+    // Parser for triple quoted strings (both single and double quotes)
+    choice((triple_quoted_parser('"'), triple_quoted_parser('\'')))
+}
+
+// TODO: not working, need to figure out how to convert the `then_with` in 0.9 to 0.10
+//
+// The commented code below shows how the 0.9 lexer handled multi-level quoted strings
+// by counting the number of opening quotes and then creating a closing delimiter
+// with the same count:
+//
+// fn quoted_string_of_quote2(
+//     quote: &char,
+//     escaping: bool,
+// ) -> impl Parser<'_, ParserInput<'_>, Vec<char>, ParserError<'_>> {
+//     let opening = just(*quote).repeated().at_least(1);
+//
+//     opening.then_with_ctx(move |opening| {
+//         if opening.len() % 2 == 0 {
+//             // If we have an even number of quotes, it's an empty string.
+//             return (just(vec![])).boxed();
+//         }
+//         let delimiter = just(*quote).repeated().exactly(opening.len());
+//
+//         let inner = if escaping {
+//             choice((
+//                 // If we're escaping, don't allow consuming a backslash
+//                 // We need the `vec` to satisfy the type checker
+//                 (delimiter.or(just(vec!['\\']))).not(),
+//                 escaped_character(),
+//                 // Or escape the quote char of the current string
+//                 just('\\').ignore_then(just(*quote)),
+//             ))
+//             .boxed()
+//         } else {
+//             delimiter.not().boxed()
+//         };
+//
+//         inner.repeated().then_ignore(delimiter).boxed()
+//     })
+// }
+
+fn quoted_string_of_quote(
+    quote: &char,
+    escaping: bool,
+    allow_multiline: bool,
+) -> impl Parser<'_, ParserInput<'_>, Vec<char>, ParserError<'_>> {
+    let q = *quote;
+
+    // Parser for non-quote characters
+    let regular_char = if allow_multiline {
+        any().filter(move |c: &char| *c != q && *c != '\\').boxed()
+    } else {
+        any()
+            .filter(move |c: &char| *c != q && *c != '\n' && *c != '\r' && *c != '\\')
+            .boxed()
```

Comment:
You can avoid boxing here by using `none_of("...")` - the types will be compatible for both branches.